lib/monaco-adapter.js

```diff
@@ -1,301 +1,498 @@
-firepad.MonacoAdapter = (function () {
-  var __bind = function (fn, me) {
-    return function () {
-      return fn.apply(me, arguments);
-    };
+'use strict';
+
+/**
+ * @function getCSS - For Internal Usage Only
+ * @param {String} clazz - CSS Class Name
+ * @param {String} bgColor - Background Color
+ * @param {String} color - Font Color
+ * @returns CSS Style Rules according to Parameters
+ */
+var getCSS = function getCSS(clazz, bgColor, color) {
+  return "." + clazz + " {\n  position: relative;\n" +
+    "background-color: " + bgColor + ";\n" +
+    "border-left: 2px solid " + color + ";\n}";
+};
+
+/**
+ * @function addStyleRule - For Internal Usage Only
+ * @desc Creates style element in document head and pushed all the style rules
+ * @param {String} clazz - CSS Class Name
+ * @param {String} css - CSS Style Rules
+ */
+var addStyleRule = function addStyleRule(clazz, css) {
+  /** House Keeping */
+  if (typeof document === 'undefined' || document === null) {
+    return false;
   }
-  var __slice = [].slice;
-
-  MonacoAdapter.prototype.ignoreChanges = false;
 
+  /** Add style rules only once */
+  if (this.addedStyleRules.indexOf(clazz) === -1) {
```

Comment:
Because I rewrote in ES6 at first, and using `[]` is not regarded as best practice. I will vouch we keep it like this, so that most of the responsibility moves toward language and library than the coder.